src/round-slider.ts

```diff
@@ -0,0 +1,525 @@
+import {
+  LitElement,
+  html,
+  css,
+  svg,
+  TemplateResult,
+  SVGTemplateResult,
+  PropertyValues,
+  CSSResultGroup,
+} from "lit";
+import { property, state } from "lit/decorators";
+
+export class RoundSlider extends LitElement {
+  @property({ type: Number }) public value: number;
+  @property({ type: Number }) public high: number;
+  @property({ type: Number }) public low: number;
+  @property({ type: Number }) public min = 0;
+  @property({ type: Number }) public max = 100;
+  @property({ type: Number }) public step = 1;
+  @property({ type: Number }) public startAngle = 135;
+  @property({ type: Number }) public arcLength = 270;
+  @property({ type: Number }) public handleSize = 6;
+  @property({ type: Number }) public handleZoom = 1.5;
+  @property({ type: Boolean }) public readonly = false;
+  @property({ type: Boolean }) public disabled = false;
+  @property({ type: Boolean, reflect: true }) public dragging = false;
+  @property({ type: Boolean }) public rtl = false;
+  @property() public valueLabel: string;
+  @property() public lowLabel: string;
+  @property() public highLabel: string;
+
+  @state() private _scale = 1;
+
+  private _reverseOrder?: boolean;
+  private _rotation?: {
+    handle: HTMLElement;
+    min: number;
+    max: number;
+    start: string;
+    type: string;
+    cooldown: number;
+  };
+
+  constructor() {
+    super();
+    this.dragEnd = this.dragEnd.bind(this);
+    this.drag = this.drag.bind(this);
+    this._keyStep = this._keyStep.bind(this);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    document.addEventListener("mouseup", this.dragEnd);
+    document.addEventListener("touchend", this.dragEnd, {
+      passive: false,
+    });
+    document.addEventListener("mousemove", this.drag);
+    document.addEventListener("touchmove", this.drag, {
+      passive: false,
+    });
+    document.addEventListener("keydown", this._keyStep);
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    document.removeEventListener("mouseup", this.dragEnd);
+    document.removeEventListener("touchend", this.dragEnd);
+    document.removeEventListener("mousemove", this.drag);
+    document.removeEventListener("touchmove", this.drag);
+    document.removeEventListener("keydown", this._keyStep);
```

Comment:
Good catch!

I think maybe all of those listeners don't actually have to be on `document`, but can be on the element so they are automatically collected on disconnect... may require some focus trickery, though...